core/backend-itwin-client/src/test/AgentAuthorizationClient.test.ts

```diff
@@ -68,47 +69,13 @@ describe("AgentAuthorizationClient (#integration)", () => {
 
   it("should get valid OIDC tokens for agent applications", async () => {
     const agentClient = new AgentAuthorizationClient(agentConfiguration);
-    const now = Date.now();
-    const jwt: AccessToken = await agentClient.getAccessToken(requestContext);
-
-    const expiresAt = jwt.getExpiresAt();
-    chai.assert.isDefined(expiresAt);
-    chai.assert.isAbove(expiresAt!.getTime(), now);
-
-    const startsAt = jwt.getStartsAt();
-    chai.assert.isDefined(startsAt);
-    chai.assert.isAtLeast(startsAt!.getTime(), expiresAt!.getTime() - 1 * 60 * 60 * 1000); // Starts at least 1 hour before expiry
+    const jwt: AccessToken = await agentClient.getAccessToken(requestContext) ?? "";
 
     await validator.validateContextRegistryAccess(jwt);
     await validator.validateIModelHubAccess(jwt);
 
-    const refreshJwt: AccessToken = await agentClient.getAccessToken(requestContext);
+    const refreshJwt: AccessToken = await agentClient.getAccessToken(requestContext) ?? "";
     await validator.validateContextRegistryAccess(refreshJwt);
     await validator.validateIModelHubAccess(refreshJwt);
   });
-
-  it("should not refresh token unless necessary", async () => {
-    const agentClient = new AgentAuthorizationClient(agentConfiguration);
-
-    const jwt: AccessToken = await agentClient.getAccessToken(requestContext);
-
-    // Refresh after a second, and the token should remain the same
-    await BeDuration.wait(1000);
-    let refreshJwt: AccessToken = await agentClient.getAccessToken(requestContext);
-    chai.assert.strictEqual(refreshJwt, jwt);
-
-    // Set the expiry of the token to be 2 min from now, and the token should remain the same
-    const twoMinFromNow = new Date(Date.now() + 2 * 60 * 1000);
-    const jwtExpiresAtTwoMinFromNow = new AccessToken(jwt.toTokenString(IncludePrefix.No), jwt.getStartsAt(), twoMinFromNow, jwt.getUserInfo());
-    (agentClient as any)._accessToken = jwtExpiresAtTwoMinFromNow;
-    refreshJwt = await agentClient.getAccessToken(requestContext);
-    chai.assert.strictEqual(refreshJwt, jwtExpiresAtTwoMinFromNow);
-
-    // Set the expiry of the token to be less than a min from now, and the token should be refreshed
-    const lessThanMinFromNow = new Date(Date.now() + 59 * 1000);
-    const jwtExpiresAtLessThanMinFromNow = new AccessToken(jwt.toTokenString(IncludePrefix.No), jwt.getStartsAt(), lessThanMinFromNow, jwt.getUserInfo());
-    (agentClient as any)._accessToken = jwtExpiresAtLessThanMinFromNow;
-    refreshJwt = await agentClient.getAccessToken(requestContext);
-    chai.assert.notStrictEqual(refreshJwt.toTokenString(IncludePrefix.No), jwtExpiresAtLessThanMinFromNow.toTokenString(IncludePrefix.No));
-  });

```

Comment:
@calebmshafer Is this test fine to remove since we can't mock out the expiration?

Reply:
I think we do need a new test to ensure token refresh is happening whenever you call `getAccessToken` but I agree that this test will not continue to work and needs to be completely re-written.

Lets do that over in the new repo when we pull it out. I have a whole list of test scenarios (currently in my head :) ) that I want us to start covering.